### PR TITLE
Add support for 'FROM scratch'

### PR DESCRIPTION
**Pull request by goldmann**
This commit adds support for building container images
using the special base iamge 'scratch'. Using this feature
requires that image is prepared in smart way and not every
module will work because the base image is empty and does
not provide any libraries.

A common use case for this type of images is to use it as
metadata images or to store binary builds (potentially
being a result of mult-stage builds).

To support these use cases a new property was added to resources:
'dest'. This defines where the artifact should be placed within the image.

https://github.com/cekit/cekit/issues/487

### CODE REVIEW

**Comment by goldmann** [comment]
This is initial take on it. Will probably require a few rounds of updates.

File: tests/images/scratch/image.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+labels:
+  - name: "io.cekit.test"
+    value: "This is a CEKit test label"
+
+envs:
+  - name: "CEKIT_TEST"
+    value: "test"
+
+artifacts:
+    # Both files will be added to a /files/ directory in teh container (which will be created)
```
Comment (kyguy) [inline tests/images/scratch/image.yaml:15]: teh -> the

Comment (goldmann) [reply tests/images/scratch/image.yaml:15]: Thanks! Will fix it in the next round! :)

**Comment by goldmann** [comment]
Documentation preview can be viewed at: https://1736-102005331-gh.circle-artifacts.com/0/docs/_build/html/handbook/scratch.html

File: cekit/descriptor/resource.py [hunk 2]
```diff
@@ -317,6 +325,7 @@ class _PathResource(Resource):
         'map': {
             'name': {'type': 'str', 'desc': 'Key used to identify the resource'},
             'target': {'type': 'str', 'desc': 'Target file name for the resource'},
+            'dest': {'type': 'str', 'desc': 'Destination directory inside of the container, should be used together with "target"', 'default': '/tmp/artifacts/'},
```
Comment (ppatierno) [inline cekit/descriptor/resource.py:328]: the first examples in the doc don't show the usage of "target" together with "dest", or?

Comment (goldmann) [reply cekit/descriptor/resource.py:328]: Correct, `target` is optional.

Comment (goldmann) [reply cekit/descriptor/resource.py:328]: I think I could reword the description in schema. Plus I need to add this to the resource documentation, so thanks!

Comment (goldmann) [reply cekit/descriptor/resource.py]: This should be now resolved.

**Pull request merged**